src/spacy/domain.clj

```diff
@@ -145,6 +153,25 @@
                    (update ::waiting-queue rest)
                    (update ::schedule conj scheduled-session))})))
 
+(defn- find-by-id [id {::keys [waiting-queue schedule]}]
+  (->> (concat waiting-queue schedule)
+       (filter #(= id (get-in % [::session ::id])))
+       first))
+
+(defn delete-session [state current-user {:keys [id]}]
+  {:post [(s/valid? ::outcome %)]}
+  (let [session (find-by-id id state)
+        is-the-sponsor? (= current-user (::sponsor session))]
+    (if (and session is-the-sponsor?)
+      {::event (-> state
+                   (update ::waiting-queue (partial remove #{session}))
```

Comment:
I think using `partial` with a set is a very clever way of creating a remove function